Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/ConceptSearch.java

```diff
@@ -148,6 +145,27 @@ private void initBinder(WebDataBinder binder) {
                 HttpStatus.OK);
     }
 
+    /**
+     * 
+     * This method is used for setting the correct values for concept search
+     * parameters.
+     * 
+     * @param conceptSearchParameters
+     * @return
+     */
+    private void preprocessConceptSearchParameterBean(ConceptSearchParameters conceptSearchParameters) {
+        if (conceptSearchParameters.getType_uri() != null) {
+            conceptSearchParameters.setType_id(uriHelper.getTypeId(conceptSearchParameters.getType_uri()));
+        }
+    }
+
+    private int getNumberOfRecordsPerPage(ConceptSearchParameters conceptSearchParameters) {
+        if (conceptSearchParameters.getNumber_of_records_per_page() != null) {
+            return conceptSearchParameters.getNumber_of_records_per_page();
+        }
+        return numberOfRecordsPerPage;
```

Comment:
same as above, this can be a one liner (ternary operator). and then it wouldn't need an extra method.